### PR TITLE
Fix save with "uint8" `dtype` for `Mask` class

### LINKED ISSUE #379: When saving mask, fails due to dtype = 'bool'
**Describe the bug**
Saving a mask fails. According to the documentation (https://geoutils.readthedocs.io/en/stable/mask_class.html), I think it should automatically convert the bool dtype to numpy.uint8.

**To Reproduce**
```
outlines_nlcd = gu.Vector(path_to_nlcd)
inlier_mask_nlcd = outlines_nlcd.create_mask(reference_demr)
inlier_mask_all = nlier_mask_nlcd
```

**Expected behavior**
The mask saves as numpy.uint8 dtype, and can be viewed in black/white.

**System (please complete the following information):**
 - OS: macos Ventura
- geoutils v0.0.12

image

### PR DESCRIPTION
**Pull request by rhugonnet**
Trying to properly force this behaviour for the `Mask` class.

Resolves #379

### CODE REVIEW

File: geoutils/raster/raster.py [hunk 1]
```diff
@@ -2189,7 +2189,6 @@ def save(
 
         :returns: None.
         """
-        dtype = self.data.dtype if dtype is None else dtype
```
Comment (adehecq) [inline geoutils/raster/raster.py:2192]: Why did you need to delete this line?

Comment (rhugonnet) [reply geoutils/raster/raster.py:2192]: Because `dtype` was never used later in the function (`mypy` strangely didn't pick that up but some of my other IDE linters did)

Comment (adehecq) [reply geoutils/raster/raster.py:2192]: Haha ok... In that case, it should be removed from the argument list. Or we could use the argument to force the output type to a preferred type. I will raise an issue.
